src/containers/SignUp/Form/Form.tsx

```diff
@@ -21,18 +24,48 @@ export default function Form() {
     gender: "",
   });
 
+  const [passwordType, setPasswordType] = useState("password");
+  const router = useRouter();
+
+  const signUpValidationResult = useMemo(() => {
+    return {
+      isEmailInvalid: !validator.email(userInfo.email),
+      isPasswordInvalid: !validator.password(userInfo.password),
+      isPhoneNumberInvalid: !validator.phoneNumber(userInfo.phoneNumber),
+    };
+  }, [userInfo]);
+
   const handleSignUpSubmit = (ev: React.FormEvent<HTMLFormElement>) => {
     ev.preventDefault();
-    console.log(userInfo);
-  };
+    /** 
+     const translate = {
+       name: "이름은",
+       email: "이메일은",
+       password: "비밀번호는",
+       phoneNumber: "전화번호는",
+       grade: "학년 선택은",
+       gender: "성별 선택은",
+     }; 
 
-  function emailCheck(ev: React.ChangeEvent<HTMLInputElement>) {
-    if (true) {
-      //Todo : validator 유효성 검사 로직 추가
-      // 유효성 검사가 필요한 항목들을 모아서 재사용 가능하게 만들것
-      setIsAvailableId(true);
+     Object.keys(userInfo).forEach((el: string, idx) => {
+       if (userInfo[el] === "" || userInfo[el] === "select") {
+         alert(`${translate[el]} 필수입니다.`);
+         return false;
+       }
+     });
+    추후 기획에서 필수값이 결정되면 작업 예정*/
+
+    if (
+      signUpValidationResult.isEmailInvalid ||
+      signUpValidationResult.isPasswordInvalid ||
+      signUpValidationResult.isPhoneNumberInvalid
+    ) {
+      return;
     }
-  }
+
+    signUp(userInfo);
+    router.replace("/");
```

Comment:
```suggestion 
    signUp(userInfo);
    router.replace("/");
```
이럼 실패해도 무조건 홈 화면으로 이동하니까 
성공했을때 `router.replace`를 실행하는게 맞는 플로우같네요